File: majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int>ans;
        int cnt1=0,cnt2=0,cnt=nums.size()/3;
        int a=INT_MIN,b=INT_MIN;
        for(auto it:nums){
            if(cnt1==0 && it!=b) {
                a=it; cnt1++;
            }
            else if(cnt2==0 && it!=a) {
                b=it; cnt2++;
            }
            else if(it==a) cnt1++;
            else if(it==b) cnt2++;
            else {
                cnt1--; cnt2--;
            }
        }
        cnt1=0; cnt2=0;
        for(auto it:nums){
            if(it==a) cnt1++;
            else if(it==b) cnt2++;
        }
        if(cnt1>cnt) ans.push_back(a);
        if(cnt2>cnt) ans.push_back(b);
        return ans;
    }
};

File: majority-element-ii/majority-element-ii.py
class Solution(object):
    def majorityElement(self, nums):
        cnt1=0 
        cnt2=0
        a = float('-inf')
        b = float('-inf')
        n=len(nums)
        for idx in nums:
            if(cnt1==0 and idx!=b): 
                cnt1=1
                a=idx
            elif(cnt2==0 and idx!=a): 
                cnt2=1
                b=idx
            elif(idx==a): cnt1+=1
            elif(idx==b): cnt2+=1
            else:
                cnt1-=1
                cnt2-=1
        cnt1=0
        cnt2=0
        for idx in nums:
            if(idx==a): cnt1+=1
            elif(idx==b): cnt2+=1
        lst=[]
        if(cnt1>n//3): lst.append(a)
        if(cnt2>n//3): lst.append(b) 
        return lst